File: CPP/program/Vehicle.cpp
#pragma once
// Baris ini memastikan bahwa file header hanya disertakan sekali dalam unit kompilasi.

#include <iostream> // Library ini digunakan untuk operasi input/output.
#include <string>   // Library ini digunakan untuk menggunakan tipe data string.

using namespace std; // Baris ini menggunakan namespace standar.

// Ini adalah deklarasi dari kelas Vehicle.
class Vehicle{
    private: 
        // Ini adalah anggota data pribadi dari kelas Vehicle.
        string plat_nomor, merk, warna;
        int tahun_produksi;

    public:
        // Ini adalah konstruktor default dari kelas Vehicle.
        Vehicle()
        {
            this->plat_nomor = "";
            this->merk = "";
            this->warna = "";
            this->tahun_produksi = 0;
        }

        // Ini adalah konstruktor parameter dari kelas Vehicle.
        Vehicle(string plat_nomor,string merk,string warna,int tahun_produksi,string jenis_Vehicle){
            this->plat_nomor = plat_nomor;
            this->merk = merk;
            this->warna = warna;
            this->tahun_produksi = tahun_produksi;
        }

        // Ini adalah metode setter untuk mengatur nilai anggota data pribadi.
        void setPlatNomor(string plat_nomor) {
            this->plat_nomor = plat_nomor;
        }

        void setMerk(string merk) {
            this->merk = merk;
        }

        void setWarna(string warna) {
            this->warna = warna;
        }

        void setTahunProduksi(int tahun_produksi) {
            this->tahun_produksi = tahun_produksi;
        }

        // Ini adalah metode getter untuk mendapatkan nilai anggota data pribadi.
        string getPlatNomor() {
            return plat_nomor;
        }

        string getMerk() {
            return merk;
        }

        string getWarna() {
            return warna;
        }

        int getTahunProduksi() {
            return tahun_produksi;
        }

        // Ini adalah destruktor dari kelas Vehicle.
        ~Vehicle()    
        {

        }
};

File: Python/program/Vehicle.py
# Baris ini memastikan bahwa kelas hanya dideklarasikan sekali dalam satu modul.

# Ini adalah deklarasi dari kelas Vehicle.
class Vehicle:
    def __init__(self, plat_nomor="", merk="", warna="", tahun_produksi=0):
        # Ini adalah konstruktor kelas Vehicle.
        self.plat_nomor = plat_nomor
        self.merk = merk
        self.warna = warna
        self.tahun_produksi = tahun_produksi

    # Ini adalah metode setter untuk mengatur nilai atribut objek.
    def set_plat_nomor(self, plat_nomor):
        self.plat_nomor = plat_nomor

    def set_merk(self, merk):
        self.merk = merk

    def set_warna(self, warna):
        self.warna = warna

    def set_tahun_produksi(self, tahun_produksi):
        self.tahun_produksi = tahun_produksi

    # Ini adalah metode getter untuk mendapatkan nilai atribut objek.
    def get_plat_nomor(self):
        return self.plat_nomor

    def get_merk(self):
        return self.merk

    def get_warna(self):
        return self.warna

    def get_tahun_produksi(self):
        return self.tahun_produksi

  